filters/src/crop_box.cpp

```diff
@@ -78,8 +78,8 @@ pcl::CropBox<pcl::PCLPointCloud2>::applyFilter (PCLPointCloud2 &output)
   for (size_t index = 0; index < indices_->size (); ++index)
   {
     // Get local point
-    int point_offset = ((*indices_)[index] * input_->point_step);
-    int offset = point_offset + input_->fields[x_idx_].offset;
+    size_t point_offset = static_cast<size_t>((*indices_)[index]) * input_->point_step;
```

Comment:
Please use `std::size_t` instead.

Also, why is the index being cast to `size_t`? Is that really required? Could you please test if this works without the explicit cast

Reply:
Thanks for your prompt reply :) What do you mean by "all occurrences"? I think I found all in `pcl::CropBox`, or did you spot another one?

Good point about the namespace, I'll add `std::`.

Regarding the type cast: `indices_` is declared as `std::vector<int>` in https://github.com/PointCloudLibrary/pcl/blob/6129a3ff78b4d12cff120142d3a18f98236f3bf4/common/include/pcl/pcl_base.h#L60
`point_step` is declared as `uint32_t` in https://github.com/PointCloudLibrary/pcl/blob/6129a3ff78b4d12cff120142d3a18f98236f3bf4/common/include/pcl/PCLPointCloud2.h#L42

This means the multiplication is carried out in 32 bit precision. By casting one of the operators to a wider type, I force the compiler to promote all operands to `size_t` and perform the calculation in 64 bit. See "Evaluating arithmetic expressions" on https://www.learncpp.com/cpp-tutorial/44-implicit-type-conversion-coercion/.

I made this example on https://onlinegdb.com/H1H0HhQYr:
```c++
#include <iostream>

using namespace std;

int main()
{
    int i = 2000000000;
    unsigned int u = 10;
    
    size_t s = i * u;
    size_t s2 = static_cast<size_t>(i) * u;
    
    cout << i << " * " << u << " = " << s << endl;
    cout << i << " * " << u << " = " << s2 << endl;
}
```
Prints:
```
2000000000 * 10 = 2820130816                                                                                                                                                                    
2000000000 * 10 = 20000000000  
```

Reply:
All occurrences of `size_t` in the lines that you touched. Thanks for the test example. I was under the belief that multiplication is done at full register and then the result is cropped later.